### PR TITLE
Added dead_letter_config

### PR DESCRIPTION
**Pull request by 3h4x**
## what
* Added variable to configure DLQ

## why
* To have DLQ in lambda

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -245,3 +245,13 @@ variable "custom_iam_policy_arns" {
   description = "ARNs of custom policies to be attached to the lambda role"
   default     = []
 }
+
+variable "dead_letter_config_target_arn" {
+  type        = string
+  description = <<EOF
+  ARN of an SNS topic or SQS queue to notify when an invocation fails. If this option is used, the function's IAM role
```
Comment (Gowiem) [inline variables.tf:252]: Good docs 👍

**Review by Gowiem — DISMISSED** [review]
@3h4x LGTM! Haven't seen you in a while -- glad I saw this one come through my email!

File: main.tf [hunk 2]
```diff
@@ -34,6 +34,10 @@ resource "aws_lambda_function" "this" {
   tags                           = var.tags
   timeout                        = var.timeout
 
+  dead_letter_config {
```
Comment (Gowiem) [inline main.tf:37]: @3h4x looks like you'll need to make this block dynamic based on the value of the variable. Tests are hitting the following: 

```
Error: Missing required argument

  on ../../main.tf line 38, in resource "aws_lambda_function" "this":
  38:     target_arn = var.dead_letter_config_target_arn

The argument "dead_letter_config.0.target_arn" is required, but no definition
was found.
}
    apply.go:15: 
        	Error Trace:	apply.go:15
        	            				examples_complete_test.go:37
        	Error:      	Received unexpected error:
        	            	FatalError{Underlying: error while running command: exit status 1; 
        	            	Error: Missing required argument
        	            	
        	            	  on ../../main.tf line 38, in resource "aws_lambda_function" "this":
        	            	  38:     target_arn = var.dead_letter_config_target_arn
        	            	
        	            	The argument "dead_letter_config.0.target_arn" is required, but no definition
        	            	was found.
        	            	}
        	Test:       	TestExamplesComplete
```

Mind doing that update?

File: main.tf [hunk 3]
```diff
@@ -34,6 +34,10 @@ resource "aws_lambda_function" "this" {
...
+    target_arn = var.dead_letter_config_target_arn
+  }
```
Comment (nitrocode) [inline main.tf:39]:
```suggestion
  dynamic "dead_letter_config" {
    for_each = try(length(var.dead_letter_config_target_arn), 0) > 0 ? [true] : []
  
    content {
      target_arn = var.dead_letter_config_target_arn
    }
  }
```

File: main.tf (hunk 2 above)
Comment (nitrocode) [reply main.tf]: I added and committed the above change

**Review by nitrocode — APPROVED** [review]

**Pull request merged**